modules/op-creating-pipeline-run-using-pipelines-as-code.adoc

```diff
@@ -0,0 +1,176 @@
++
+[source,yaml]
+----
+...
+  pipelinesascode.tekton.dev/on-cel-expression: |
+    event == "pull_request" && "docs/\*.md".pathChanged() <1>
+...
+----
+<1> Matches all markdown files in the `docs` directory.
+
+* To match all pull requests starting with the title `[DOWNSTREAM]`:
++
+[source,yaml]
+----
+...
+  pipelinesascode.tekton.dev/on-cel-expression: |
+    event == "pull_request && event_title.startsWith("[DOWNSTREAM]")
+...
+----
+
+* To run a pipeline on a `pull_request` event, but skipping the `experimental` branch:
++
+[source,yaml]
+----
+...
+  pipelinesascode.tekton.dev/on-cel-expression: |
+    event == "pull_request" && target_branch != experimental"
+...
+----
+
+For advanced CEL based filtering while using {pac}, you can use the following fields and suffix functions:
+
+* `event`: A `push` or `pull_request` event.
+* `target_branch`: The target branch.
+* `source_branch`: The branch of origin of a `pull_request` event. For `push` events, it is same as the `target_branch`.
+* `event_title`: Matches the title of the event, such as the commit title for a `push` event, and the title of a pull or merge request for a `pull_request` event. Currently, only GitHub, Gitlab, and BitbucketCloud are the supported providers.
+* `.pathChanged`: A suffix function to a string. The string can be a glob of a path to check if the path has changed. Currently, only GitHub and Gitlab are supported as providers.
+
+[discrete]
+.Using the temporary Github App token for Github API operations
+
+You can use the temporary installation token generated by {pac} from Github App to access the Github API. The token value is stored in the temporary `{{git_auth_secret}}` dynamic variable as generated for private repositories in the `git-provider-token` key.
+
+For example, to add a comment to a pull request, you can use the `github-add-comment` task from {tekton-hub} using a {pac} annotation:
+
+[source,yaml]
+----
+...
+  pipelinesascode.tekton.dev/task: "github-add-comment"
+...
+----
+
+You can then add a task to the `tasks` section or `finally` tasks in the pipeline run definition:
+
+[source,yaml]
+----
+[...]
+tasks:
+  - name:
+      taskRef:
+        name: github-add-comment
+      params:
+        - name: REQUEST_URL
+          value: "{{ repo_url }}/pull/{{ pull_request_number }}" <1>
+        - name: COMMENT_OR_FILE
+          value: "Pipelines as Code IS GREAT!"
+        - name: GITHUB_TOKEN_SECRET_NAME
+          value: "{{ git_auth_secret }}"
+        - name: GITHUB_TOKEN_SECRET_KEY
+          value: "git-provider-token"
+...
+----
+<1> By using the dynamic variables, you can reuse this snippet template for any pull request from any repository.
+
+[NOTE]
+====
+On GitHub apps, the generated installation token is available for 8 hours, and scoped to the repository from here the events originate, unless configured differently on the cluster.
```

Comment:
```suggestion
On GitHub Apps, the generated installation token is available for 8 hours and scoped to the repository from where the events originate unless configured differently on the cluster.
```

Reply:
@sounix000  Change all incorrect instances to GitHub App/GitHub Apps as appropriate. This is a global comment. Please run it through all the files in this PR and rectify it as required.

Reply:
I've checked all the 22 files in this PR and corrected all the instances.